source/codegen/metadata/nixnetsocket/functions.py

```diff
@@ -85,206 +96,115 @@
-                'type': 'nxin_addr'
+                'grpc_type': 'AddrInfoHint',
+                'pointer': True,
+                'type': 'nxaddrinfo'
             },
             {
                 'direction': 'out',
-                'name': 'address',
-                'return_value': True,
-                'type': 'char[]'
-            },
+                'name': 'res',
+                'supports_standard_copy_convert': True,
+                'supports_standard_output_allocation': True,
+                'additional_arguments_to_output_allocation': ['library_'],
+                'pointer': True,
+                'type': 'nxaddrinfo'
+            }
         ],
-        'returns': 'char*'
+        'returns': 'int32_t'
     },
-    'InetNToP': {
-        'cname': 'nxinet_ntop',
-        'status_expression': 'address ? 0 : -1',
+    'GetLastErrorNum': {
+        'codegen_method': 'private',
+        'cname': 'nxgetlasterrornum',
+        'parameters': [],
+        'returns': 'int32_t'
+    },
+    'GetLastErrorStr': {
+        'codegen_method': 'private',
+        'cname': 'nxgetlasterrorstr',
+        'status_expression': 'error ? 0 : -1',
         'parameters': [
-            {
-                'direction': 'in',
-                'name': 'stack_ref',
-                'type': 'nxIpStackRef_t'
-            },
-            {
-                'direction': 'in',
-                'hardcoded_value': 'get_address_family(request->src().addr_case())',
-                'include_in_proto': False,
-                'name': 'af',
-                'type': 'int32_t'
-            },
-            {
-                'direction': 'in',
-                'name': 'src',
-                'grpc_type': 'Addr',
-                'pointer': True,
-                'supports_standard_copy_convert': True,
-                'type': 'void'
-            },
             {
                 'direction': 'out',
-                'name': 'dst',
                 'include_in_proto': False,
+                'name': 'buf',
                 'size': {
-                    'mechanism': 'fixed',
-                    'value': 'nxINET6_ADDRSTRLEN'
+                    'mechanism': 'passed-in',
+                    'value': 'bufLen'
                 },
                 'type': 'char[]'
             },
             {
                 'direction': 'in',
-                'name': 'size',
-                'hardcoded_value': 'nxINET6_ADDRSTRLEN',
-                'include_in_proto': False,
-                'type': 'nxsocklen_t'
+                'name': 'bufLen',
+                'type': 'size_t'
             },
             {
                 'direction': 'out',
-                'name': 'address',
+                'name': 'error',
                 'return_value': True,
                 'type': 'char[]'
             },
         ],
-        'returns': 'const char*'
+        'returns': 'char*'
     },
-    'InetPToN': {
-        'cname': 'nxinet_pton',
+    'GetPeerName': {
```

Comment:
This should be swapped with `GetNameInfo` below.

Reply:
good catch switched it up